src/modules/form/util/formUtil.ts

```diff
@@ -295,10 +295,15 @@ export const updateFormItem = (
       }
     }
 
-    throw new Error(`Link ID  ${newItem.linkId} not found`);
+    return false;
+  }
+
+  // Recurse through the item tree and replace the item
+  if (!recursiveReplace(copy.item)) {
+    // If recursiveReplace returns false, we didn't find the link ID, so this is a new element
+    copy.item = [...copy.item, newItem];
```

Comment:
I think maybe this `updateFormItem` function should go in a util function inside the `formBuilder` module. We already have soo many util functions for rendering the forms, it might be wise to keep the formBuilder utils separate.